src/main/java/com/exactpro/th2/codec/grpc/ProtoDecoder.java

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.exactpro.th2.codec.grpc;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import com.exactpro.th2.common.grpc.Direction;
+import com.exactpro.th2.common.grpc.Message;
+import com.exactpro.th2.common.grpc.RawMessage;
+import com.exactpro.th2.common.grpc.RawMessageMetadata;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.apache.commons.io.FileUtils;
+
+public class ProtoDecoder {
+	public static final String GRPC_CALL = "GRPC_CALL";
+	private static final String TEMP_DIR = "gen/test/temp";
+
+	private final Map<String, Descriptors.ServiceDescriptor> serviceDescriptorMap = new HashMap<>();
+	
+	public ProtoDecoder(Path protosPath) throws IOException {
+		Collection<File> protos = FileUtils.listFiles(protosPath.toFile(), new String[]{"proto"}, true);
+		List<ProtoSchema> protoSchemas = new ProtobufParser(TEMP_DIR).parseProtosToSchemas(protosPath.toFile(), protos);
+		protoSchemas.forEach(schema -> serviceDescriptorMap.putAll(schema.getServices()));
+	}
+	
+	public Message decode(RawMessage message) throws InvalidProtocolBufferException, JsonProcessingException {
```

Comment:
Maybe return message builder to avoid rebuild message when metadata will be added